pkg/sanity/resources.go

```diff
@@ -26,19 +27,62 @@ import (
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
 
+	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
-// VolumeInfo keeps track of the information needed to delete a volume.
-type VolumeInfo struct {
+// managedResourceInfos is a map of managed resource infos keyed by resource
+// IDs.
+type resourceInfos map[string]resourceInfo
```

Comment:
Alternatively, this could also be typed as `[]resourceInfo` which would make the LIFO property more natural/obvious. On the downside, it would also bring extra effort to remove items from the slice as we clean up resources (volumes, snapshots) during `Cleanup`.

I decided to go with a map and introduced an `order` field to keep track of when resources where added to the list.

Reply:
This uses the volume ID and snapshot ID as key, right? Is is guaranteed that those never conflict?

> it would also bring extra effort to remove items from the slice as we clean up resources (volumes, snapshots) during Cleanup

That extra effort is just finding the item and then creating a slice without it, right? That doesn't seem that hard.

Without having seen what it looks like, I suspect that `[]resourceInfo` will be the better solution.

Reply:
Made the change. I think you were right, this looks much simpler.